src/resource.ts

```diff
@@ -279,12 +283,12 @@ export default class Resource<T = any> {
    * This function can also follow templated uris. You can specify uri
    * variables in the optional variables argument.
    */
-  follow(rel: string, variables?: object): FollowablePromise {
+  follow(rel: string, variables?: object, anchor?: string): FollowablePromise {
```

Comment:
This is just an idea... instead of an API like this:

```typescript
resource.follow('author',  undefined, '#subresource');
```

We would use an API such as:

```typescript
resource.go('#subresource').follow('author');
```

Not sure on the `go` bit, but the idea is that you just get a new resource relative to the current one. If it used the fragment identifier, the new resource inherits the anchored links. Not sure about this, but just putting it out there.